homeassistant/components/sensor/bbox.py

```diff
@@ -0,0 +1,144 @@
+"""
+import logging
+from datetime import timedelta
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (BANDWIDTH_MEGABITS_SECONDS, CONF_NAME,
+                                 CONF_MONITORED_VARIABLES, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+REQUIREMENTS = ['pybbox==0.0.5-alpha']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Powered by Bouygues Telecom"
+DEFAULT_NAME = 'Bbox'
+
+# Sensor types are defined like so:
+# Name, unit, icon
+SENSOR_TYPES = {
+    'down_max_bandwidth': ['Maximum Download Bandwidth',
+                           BANDWIDTH_MEGABITS_SECONDS, 'mdi:download'],
+    'up_max_bandwidth': ['Maximum Upload Bandwidth',
+                         BANDWIDTH_MEGABITS_SECONDS, 'mdi:upload'],
+    'current_down_bandwidth': ['Currently Used Download Bandwidth',
+                               BANDWIDTH_MEGABITS_SECONDS, 'mdi:download'],
+    'current_up_bandwidth': ['Currently Used Upload Bandwidth',
+                             BANDWIDTH_MEGABITS_SECONDS, 'mdi:upload'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_VARIABLES):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+# pylint: disable=too-many-arguments
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Bbox sensor."""
+    # Create a data fetcher to support all of the configured sensors. Then make
+    # the first call to init the data.
+    try:
+        bbox_data = BboxData()
+        bbox_data.update()
+    except requests.exceptions.HTTPError as error:
+        _LOGGER.error(error)
+        return False
+
+    name = config.get(CONF_NAME)
+
+    sensors = []
+    for variable in config[CONF_MONITORED_VARIABLES]:
+        sensors.append(BboxSensor(bbox_data, variable, name))
+
+    add_devices(sensors)
+
+
+class BboxSensor(Entity):
+    """Implementation of a Bbox sensor."""
+
+    def __init__(self, bbox_data, sensor_type, name):
+        """Initialize the sensor."""
+        self.client_name = name
+        self.type = sensor_type
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._icon = SENSOR_TYPES[sensor_type][2]
+        self.bbox_data = bbox_data
+        self._state = None
+
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return round(self._state, 2)
```

Comment:
I think rouding and divided by 1000 should both be in the state method because the update purpose is just to refresh data and not not make some transformation on it.

 But I'm maybe wrong. Can you add some details on why this should be at one place or not ? (It will be useful for me for future development)